diff --git a/app/pc/eigdeflate.c b/app/pc/eigdeflate.c
index 87d6b6d..477e255 100644
--- a/app/pc/eigdeflate.c
+++ b/app/pc/eigdeflate.c
@@ -173,7 +173,7 @@ PetscErrorCode PCDestroy_eigdeflate(PC pc) {
     PetscFunctionReturn(0);
 }
 
-PetscErrorCode PCSetFromOptions_eigdeflate(PetscOptionItems* PetscOptionsObject, PC pc) {
+PetscErrorCode PCSetFromOptions_eigdeflate(PC pc, PetscOptionItems* PetscOptionsObject) {
     PC_eigdeflate* ctx;
     PetscInt M;
     PetscBool flg;
@@ -203,14 +203,14 @@ PetscErrorCode PCView_eigdeflate(PC pc, PetscViewer viewer) {
     PC_eigdeflate* ctx = (PC_eigdeflate*)pc->data;
 
     PetscFunctionBegin;
-    PetscViewerASCIIPrintf(viewer, "num. eigenvectors: %D\n", ctx->nev);
+    PetscViewerASCIIPrintf(viewer, "num. eigenvectors: %d\n", ctx->nev);
     PetscViewerASCIIPrintf(viewer, "emin: %+1.4e\n", ctx->e_min);
     PetscViewerASCIIPrintf(viewer, "emax: %+1.4e\n", ctx->e_max);
     PetscViewerASCIIPrintf(viewer, "Randomized eigenvalue calculation\n", ctx->nev_oversample);
-    PetscViewerASCIIPrintf(viewer, "over sampling: %D\n", ctx->nev_oversample);
-    PetscViewerASCIIPrintf(viewer, "power iterations: %D\n", ctx->power_its);
-    PetscViewerASCIIPrintf(viewer, "pre-smooth iterations:  %D\n", ctx->npre);
-    PetscViewerASCIIPrintf(viewer, "post-smooth iterations: %D\n", ctx->npost);
+    PetscViewerASCIIPrintf(viewer, "over sampling: %d\n", ctx->nev_oversample);
+    PetscViewerASCIIPrintf(viewer, "power iterations: %d\n", ctx->power_its);
+    PetscViewerASCIIPrintf(viewer, "pre-smooth iterations:  %d\n", ctx->npre);
+    PetscViewerASCIIPrintf(viewer, "post-smooth iterations: %d\n", ctx->npost);
     if (ctx->npre > 0 || ctx->npost > 0) {
         PetscViewerASCIIPrintf(viewer, "optimal relaxation: %+1.4e\n", ctx->alpha);
     }
@@ -228,7 +228,7 @@ PetscErrorCode PCCreate_eigdeflate(PC pc) {
     PC_eigdeflate* edef;
 
     PetscFunctionBegin;
-    CHKERRQ(PetscNewLog(pc, &edef));
+    PetscCall(PetscNew(&edef));
     pc->data = (void*)edef;
 
     edef->reig = NULL;
diff --git a/src/util/Schema.h b/src/util/Schema.h
index 870704f..57d7c22 100644
--- a/src/util/Schema.h
+++ b/src/util/Schema.h
@@ -6,6 +6,7 @@
 #include <stdexcept>
 #include <toml.hpp>
 
+#include <algorithm>
 #include <charconv>
 #include <functional>
 #include <limits>
